packages/js/data/src/crud/resolvers.ts

```diff
@@ -63,21 +67,69 @@ export const createResolvers = ( {
 
 		try {
 			const path = getRestPath( namespace, {}, urlParameters );
-			const { items }: { items: Item[] } = yield request<
-				ItemQuery,
-				Item
-			>( path, resourceQuery );
+			const { items, totalCount }: { items: Item[]; totalCount: number } =
+				yield request< ItemQuery, Item >( path, resourceQuery );
 
+			yield getItemsTotalCountSuccess( query, totalCount );
+			yield dispatch(
+				storeName,
+				'finishResolution',
+				`get${ pluralResourceName }TotalCount`,
+				[ query ]
+			);
 			yield getItemsSuccess( query, items, urlParameters );
+			for ( const i of items ) {
+				if ( i.id ) {
+					yield dispatch(
+						storeName,
+						'finishResolution',
+						`get${ resourceName }`,
+						[ i.id ]
+					);
+				}
+			}
 			return items;
 		} catch ( error ) {
+			yield getItemsTotalCountError( query, error );
 			yield getItemsError( query, error );
 			throw error;
 		}
 	};
 
+	const getItemsTotalCount = function* ( query?: Partial< ItemQuery > ) {
+		const totalsQuery = {
+			...( typeof query === 'string' ? {} : query || {} ),
```

Comment:
I don't understand two things about this `typeof query === 'string'` check (comment applies to other parts of the code that do a similar check as well)...

* Why are we checking the type of `query` at all, when we type `query` to be `Partial< ItemQuery >` in the params? Won't it always be `Partial< ItemQuery >` or `null`?
* If it does somehow get passed as a `string`, why do we just set things to an empty object?

Reply:
Yeah, it confused me at first to, and logic does indeed make it a bit more confusing.
It has to do with this [`applyNamespace` function](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/data/src/crud/utils.ts#L91-L98) that at-least the selectors get wrapped with, I should actually double check if the same thing happens for resolvers, otherwise the above won't be necessary.
The `applyNamespace` function tags on the `namespace` prop at the end of each function call.
This could mean if a user were to call `getItemsTotalCount()`, so without any parameters, the first parameter will be the `namespace` (which is a string). 
Although I only see the namespace being used in the `getItem` selectors so maybe the `applyNamespace` wrapper can be removed from all the other selectors -> https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/data/src/crud/selectors.ts#L146-L162

cc: @joshuatf as he wrote this I believe.

Reply:
I end up adding a `defaultArgs` option to the `applyNamespace`, so this way the last param isn't always overwritten by the namespace. See here: https://github.com/woocommerce/woocommerce/pull/34613/commits/09ef544211684339f8976e604fdb61829d276c3e

Reply:
I honestly still find this whole logic super confusing.

Reply:
Yeah, it is a tad confusing I do agree.
The main problem this was set out to solve is to allow the crud store to support REST endpoints that rely on a parent.
Attribute terms is a good example that lives under this url: `products/attributes/<attribute_id>/terms/<id>` and I believe the the way it's stored in Redux is by prefixing the attribute_id before the term id.

Anyway it could be worth revisiting this logic at some point, as we could probably do something similar as what the resolvers do, by the use of the `createResolvers` function, so the `namespace` prop is more globally available instead of needing to be passed in through the use of `applyNamespace`.